### ISSUE TITLE
CSX: problem with liveplot

### ISSUE THREAD

**Issue by cmazzoli** [NSLS-II/Bug-Reports#88]
In the following macro = piece of python code, essentially containing the following for loop..

```
[for kk in arange(0,10):
    movr(es_diag1_y,0.2)
    sleep(1)
    asc(pgm_en,930,950,20)
    sleep(1)
    mov(pgm_en,gs.PS.max[0]-2.0)
    sleep(1)
    ct(num=1000*(1+kk))]
```

If I start the above macro directly from the command line of my collection profile ipython session by
%run -i ~/Beamline/Macros/2016_03_04_mono_stability.py
I get no figures from liveplot for any of the scan/ct commands contained above.

On the contrary, if at the start of the collection section I give gs.TABLE = [bla bla]; gs.PLOT_Y = 'bla'
then start two dummy scan in the same vars ( asc(pgm_en,930,931,2) and ct(num=3) ), then I get the two figures and once started the macro liveplot updates them correctly.

**Comment by tacaswell** [comment]
If you define the function

``` python
def stability_scan():
    for kk in arange(0,10):
        movr(es_diag1_y,0.2)
        sleep(1)
        asc(pgm_en,930,950,20)
        sleep(1)
        mov(pgm_en,gs.PS.max[0]-2.0)
        sleep(1)
        ct(num=1000*(1+kk)
```

And then iststead of doing `%run` you do `stabliity_scan()` does it work properly?

**Comment by cmazzoli** [comment]
Hi Tom,
Thanks. I am not at the beanline anymore, Andi will take care later when
she will get in..
Thank you again.
Claudio

**Comment by cmazzoli** [comment]
Hi Tom,
I know that this is probably not the good thread, however I thought that it might be of interest for you.
I checked a long motor movement and got the following:

```
In [66]: wh_pos(gamma)

+------------+----------+-----------+------------+
| Positioner |    Value | Low Limit | High Limit |
+------------+----------+-----------+------------+
| gamma      | 10.00000 |  -3.66000 |  181.00000 |
+------------+----------+-----------+------------+

In [67]: %time movr(gamma,20); wh_pos(gamma);sleep(1);wh_pos(gamma);

   gamma              

    24.9155 deg       


+------------+----------+-----------+------------+
| Positioner |    Value | Low Limit | High Limit |
+------------+----------+-----------+------------+
| gamma      | 24.91550 |  -3.66000 |  181.00000 |
+------------+----------+-----------+------------+

+------------+----------+-----------+------------+
| Positioner |    Value | Low Limit | High Limit |
+------------+----------+-----------+------------+
| gamma      | 25.41170 |  -3.66000 |  181.00000 |
+------------+----------+-----------+------------+
CPU times: user 1.68 s, sys: 172 ms, total: 1.86 s
Wall time: 31 s

In [68]: wh_pos(gamma)

+------------+----------+-----------+------------+
| Positioner |    Value | Low Limit | High Limit |
+------------+----------+-----------+------------+
| gamma      | 29.99040 |  -3.66000 |  181.00000 |
+------------+----------+-----------+------------+

In [69]: wh_pos(gamma)

+------------+----------+-----------+------------+
| Positioner |    Value | Low Limit | High Limit |
+------------+----------+-----------+------------+
| gamma      | 30.00000 |  -3.66000 |  181.00000 |
+------------+----------+-----------+------------+

In [70]: wh_pos(gamma)

+------------+----------+-----------+------------+
| Positioner |    Value | Low Limit | High Limit |
+------------+----------+-----------+------------+
| gamma      | 30.00000 |  -3.66000 |  181.00000 |
+------------+----------+-----------+------------+

In [71]: 

```

as Ken was suggesting. I would say that we should get a check on the flag + if the motor is updating in the direction and increment requested/reasonable (to avoid jittering and other instabilities) before starting a countdown and declaring a problem with the motor if the motion doesn't return "done" before.
Please let me know your thoughts about.

**Comment by cmazzoli** [comment]
Second test Tom.
I tried with a different (but similar) "macro" (see below) and by defining a function as you suggested.
Same result = LivePlot seems unable to start the plot correctly. But if it was previously open, then no problem.

Function version..

```
def EPU2_pol_check():
    #for kk in arange(-180,181,18):
    for kk in arange(-180,181,36):
        mov(eta,kk)
        sleep(2)
        mov(eta,kk)
        sleep(2)
        mov(eta,kk)
        sleep(2)
        asc(pgm_en,970,1030,60)
        olog('EPU2 mode{}, phase {:.2f}mm and gap {:.3f}mm'.format(caget('SR:C23-ID:G1A{EPU:2-Ax:Phase}Phs:Mode-RB'),epu2.phase.readback.value,epu2.gap.readback.value))
```

**Comment by cmazzoli** [comment]
PS: once LivePlot is broken I have to restart the ipython session to regain control on the plot windows..

**Comment by tacaswell** [comment]
The moving issues here may also be due to the move timing out at 30s,

Try `%time movr(gamma, 20, timeout=None); wh_pos(gamma);sleep(1);wh_pos(gamma);`

**Comment by tacaswell** [comment]
Can you reproduce the issue with the plots with out using `%run`?  Can you show the full transcript of a session where this broke?  There is too much state involved in this system to be able to understand what is going on from the information provided.

**Comment by cmazzoli** [comment]
Sorry, I thought that I had already written this to you but it was left on the screen..

PPS: the above definition of the function was always done in a file, read through %run -i ... I don't know how to do it differently. Apologies..

About the timeout point: I still haven't tried but it is SURELY connected with the f**king timeout!!
Have you read the end of my previous comment/suggestion above??

**Comment by ambarb** [comment]
If I copy and paste from a text file to bluesky (instead of %run), LivePlot seems to work just fine (except for fccd_stats, but that is a separate issue).

**Comment by klauer** [comment]
By default, moves do not time out at 30 seconds any more. Additionally it appears that the issue was side-stepped by not using `%run`. Please re-open with further details should the problem persist.
